### PR TITLE
Remove bus from datasource api

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:
We need to add the interface DatasourcePermissionsService in enterprise. Fix here on this PR https://github.com/grafana/grafana-enterprise/pull/2831.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/datasource_permissions.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package api
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/models"
+)
+
+type DatasourcePermissionsService interface {
```
Comment (spinillos) [inline pkg/api/datasource_permissions.go:9]: We usually put the OSS prefix on classes that are going to be override in Enterprise. This way indicates that any change here, affects to Enterprise.

**Review by spinillos — APPROVED** [review]
LGTM

File: pkg/api/datasource_permissions.go (hunk 1 above)
Comment (zserge) [reply pkg/api/datasource_permissions.go:9]: But it's an interface, it shouldn't matter, or I'm mistaken? `DatasourcePermissionsServiceImpl` could be `OSSDatasourcePermissionsService` if that's preferred, but the interfaces I'd assume to be under a common name?

**Review by zserge — APPROVED** [review]

File: pkg/api/datasource_permissions.go (hunk 1 above)
Comment (spinillos) [reply pkg/api/datasource_permissions.go:9]: Oh sorry, I clicked in the wrong line. Yes, it's `DatasourcePermissionsServiceImpl`

**Review by spinillos — APPROVED** [review]

**Review by zserge — APPROVED** [review]

**Pull request merged**